flocker/control/_protocol.py

```diff
@@ -81,7 +88,8 @@ class ClusterStatusCommand(Command):
     in the convergence agent during startup.
     """
     arguments = [('configuration', DeploymentArgument()),
-                 ('state', DeploymentArgument())]
+                 ('state', DeploymentArgument()),
+                 ('eliot_context', Unicode())]
```

Comment:
This should be a simple refactoring.  A `Unicode` subclass with `toString` and `fromString` overridden to call the relevant eliot APIs - `serialize_task_id` and `continue_task`.  The eliot `Logger` object is maybe a little tricky...   Possibly this will need `toStringProto` and `fromStringProto` instead, in order to have access to the `ControlAMP` instance (which should be sufficient to reach the right `Logger` instance).  Relying on the protocol instance to provide this information is not really ideal - but that's what this version of the code does too, so at least it's not really worse.

Another option would be to have a module-scope logger object but maybe that's worse (since it will be shared between more things and since it's global state that the tests will have to replace and then restore).